setup.cfg

```diff
@@ -1,3 +1,21 @@
+[metadata]
+name = matrix-synapse-ldap3
+url = https://github.com/matrix-org/matrix-synapse-ldap3
+version = attr: ldap_auth_provider.__version__
+description = An LDAP3 auth provider for Synapse
+long_description = file: README.rst
```

Comment:
I'm really liking `attr:` and `file:` rather than our (nice, but not really belonging in a config file) `exec_file` and `read_file`.

(I also believe I read somewhere that setuptools doesn't actually execute the code, it reads it from the AST.)